client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -224,12 +224,32 @@ public void flush() {
                         // available writers.
                         success = false;
                         log.warn("Flush on segment {} failed due to {}, it will be retried.", writer.getSegmentName(), e.getMessage());
+                        tryWaitForSuccessors();
                         break;
                     }
                 }
             }
         }
     }
+    
+    /**
+     * This is used by flush to optimistically wait for the `handleLogSealed` work to be completed
+     * to prevent a busy loop while waiting for the segments to be updated following a sealed
+     * segment. Please note there are no guarantees about when this method returns. It can be
+     * thought of as a sleep.
+     */
+    @GuardedBy("writeFlushLock")
+    private void tryWaitForSuccessors() {
+        if (sealedSegmentQueue.isEmpty()) {
+            synchronized (writeSealLock) {
+                // Nothing needs to be done here.
```

Comment:
Are  you sure that the JVM (or even javac) won't drop this block? AFAIK the JVM is quite aggressive in skipping useless stuff and monitor related code

Reply:
Yeah this could be a possibility. 

Could we try to use a ReusableLatch here or some Semaphore (or any other synchronization primitive)?

Reply:
probably if you add a metric here (just a counter that counts how many times we fall into this case) the JVM will be happy as it thinks that we are doing something useful

Reply:
I think SpotBugs doesn't like this either. The build failed due to that.